bsip-0061.md

```diff
@@ -0,0 +1,169 @@
+    BSIP: TBD
+    Title: Fix locked accounts with circular dependencies
+    Authors: OpenLedgerApp <https://github.com/OpenLedgerApp>
+    Status: Draft
+    Type: Protocol
+    Created: 2019-02-22
+    Updated: 2019-03-01
+    Discussion: https://github.com/bitshares/bsips/issues/94
+    Worker:
+
+
+# Abstract
+Every account has two authorities assigned: owner and active.
+
+An authority is a set of keys and/or accounts, each of which is assigned a weight.
+Each authority has a weight threshold that must be crossed before an action requiring that authority may be performed.
+The owner authority is designed for cold-storage, and its primary role is to update the active authority or to change the owner authority.
+The active authority is meant to be a hot key and can perform any action except changing the owner authority.
+
+BitShares has locked accounts with circular dependencies. Some of them are locked by mistake. So, locked accounts couldn't buy/sell their assets, make transfer and e. c.
+
+The community has had extensive discussions about how to resolve this issue:
+
+    1. To add additional functionality "avoid circular dependencies"to the nearest hard fork;
+    2. To find all locked accounts with circular dependencies and undo the last account_update operation.
+
+There are some cases that led to locked account:
+
+    - Alice updates, via account_update_operation, her active and owner authority to herself;
+    - Bob updates, via account_update_operation, his active and owner authority to a locked account;
+    - Jill updates, via account_update_operation, her active and owner authority threshold greater than a sum of authorities weights.
+
+This proposal is to implement a hard-fork that would prevent it in the future and to make posible unlock locked accounts.
+New operation unlock_account will be added, account_update_evaluator and account_create_evaluator will be modified.
```

Comment:
IMO it does not make sense to add an operation that will at most be used by the few accounts that are locked now. Since the goal is to prevent such locks in the future, the operation would be useless once the accounts have been unlocked.
It is much simpler to automatically unlock them at hardfork time.

Reply:
According to this post (https://github.com/bitshares/bsips/issues/94#issuecomment-412261693), we have proposed options. We can only leave an automatic approach if the community agrees.

Reply:
Oh, nice catch. To (allegedly) quote our former German chancellor Adenauer - "Why should I care about the silly things I said yesterday?" :-)

I think the question is relevant to the cost/benefit discussion. Perhaps the community should decide, like you said.